jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -259,6 +264,16 @@ private Span setTagAsObject(String key, Object value) {
     return this;
   }
 
+  /**
+   * uses {@link #setTagAsObject} to only insert tags that don't have special significance
+   */
+  private void sanitizeAndSetTags(Map<String, Object> tags) {
```

Comment:
Also, once we upgrade this to jaeger.thrift, all this special handling goes away.